perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/UsersManagerBlImpl.java

```diff
@@ -706,10 +706,10 @@ private UserExtSource getUserExtSourceByUniqueAttributeValue(PerunSession sess,
 	}
 
 	@Override
-	public UserExtSource getUserExtSourceFromMultipleIdentifiers(PerunSession sess, PerunPrincipal principal) throws InternalErrorException, UserExtSourceNotExistsException {
+	public UserExtSource getUserExtSourceFromMultipleIdentifiers(PerunSession sess, PerunPrincipal principal) throws InternalErrorException, UserExtSourceNotExistsException, WrongReferenceAttributeValueException {
 		String additionalIdentifiers = principal.getAdditionalInformations().get(additionalIdentifiersAttributeName);
 		if (additionalIdentifiers == null) {
-			throw new UserExtSourceNotExistsException("Mandatory attribute is not defined: ".concat(additionalIdentifiersAttributeName));
+			throw new WrongReferenceAttributeValueException("Principal's mandatory attribute is not defined: ".concat(additionalIdentifiersAttributeName));
```

Comment:
Are you sure this is not rather a configuration error? That would mean using InternalErrorException.

Reply:
InternalErrorException would be definitely better here. This is probably not attribute as we are using it in Perun. It is more like missing information.

Reply:
Changed to InternalErrorexception.

Reply:
What about reworking the exception message as well? I will probablly have hard time to figure out that I have to change the config file after I will find this message in logs.

Reply:
Done